### ISSUE TITLE
LiveData.delay

### ISSUE THREAD

**Issue by sczerwinski** [sczerwinski/android-lifecycle#11]
Implement `delayStart` operator for `LiveData`.

```kotlin
val delayedLiveData = liveData.delayStart(timeInMillis = 1000L)

val delayedLiveData = liveData.delayStart(timeInMillis = 1000L, context = viewModelScope.coroutineContext)
```

**Comment by sczerwinski** [comment]
This won't guarantee the order, but it's probably a good start:
```kotlin
fun <T> LiveData<T>.delay(
    timeInMillis: Long,
    context: CoroutineContext = EmptyCoroutineContext
): LiveData<T> {
    val result = MediatorLiveData<T>()
    result.addSource(this) { value ->
        val source = liveData(context) {
            delay(timeInMillis)
            emit(value)
            result.removeSource(source)
        }
        result.addSource(source) { x -> result.value = x }
    }
    return result
}
```

**Comment by sczerwinski** [comment]
With `LiveData`, which only remembers latest emitted value, delaying each element makes no sense. It is probably more reasonable to use `throttleWithTimeout()` instead.

However, it might be a good idea to delay start of emitting values, i.e. throttle, but only for first element(s).

**Comment by sczerwinski** [comment]
Implemented: 3cc62cbf
